xtask/src/tools/lychee.rs

```diff
@@ -63,29 +61,31 @@ impl LycheeRunner {
         let lychee_client = self.client.clone();
 
         rt.block_on(async move {
-            let links = Collector::new(None)
+            let links: Vec<Request> = Collector::new(None)
                 .collect_links(inputs)
                 .await
                 .collect::<LycheeResult<Vec<_>>>()
                 .await?;
 
-            let mut failed_checks: Vec<Uri> = vec![];
-            let links_size = links.len();
+            let failed_link_futures: Vec<_> = links
+                .into_iter()
+                .map(|link| tokio::spawn(get_failed_request(lychee_client.clone(), link)))
+                .collect();
+
+            let links_size = failed_link_futures.len();
 
-            for link in links {
-                let response = lychee_client.check(link).await?;
-                if response.status().is_failure() {
-                    failed_checks.push(response.1.uri.clone());
-                } else if response.status().is_success() {
-                    println!("[✓] {}", response.1.uri.as_str());
+            let mut failed_checks = Vec::with_capacity(links_size);
+            for f in failed_link_futures.into_iter() {
+                if let Some(failure) = f.await.expect("unexpected error while processing links") {
```

Comment:
and here i'm awaiting for all of those futures one by one

Reply:
This is a sort of ˋjoin` with all prepared futures 👍🏻